source/ShipInfoPanel.cpp

```diff
@@ -449,14 +466,9 @@ void ShipInfoPanel::DrawWeapons(const Rectangle &bounds)
 	int turretRows = max(count[0][1], count[1][1]);
 	// If there are both guns and turrets, add a gap of ten pixels.
 	double height = 20. * (gunRows + turretRows) + 10. * (gunRows && turretRows);
+	bool pageNeeded = height > (sprite->Height() * scale);
+	height = pageNeeded ? sprite->Height() * scale : height;
```

Comment:
Why not use the size of the actual bounding box here instead of multiplying the sprite height by scale?
For a particularly wide and short sprite, this will result in this value you're comparing to being much lower than the actual available space; I don't think that's what we want.